encoding/encoding.go

```diff
@@ -64,6 +64,12 @@ type StreamSerializer interface {
 	// StreamRequest creates a root stream request, a stream request reader using
 	// body reader provided
 	StreamRequest(body io.Reader) (*transport.Request, StreamRequestReader, error)
+
+	// IsClientStreaming returns true for client streaming methods
+	IsClientStreaming() bool
+
+	// IsServerStreaming returns true for server streaming methods
+	IsServerStreaming() bool
```

Comment:
I like this! Way simpler and invasive!

Did you think of having a a method `Type() type` in the Serializer interface?
We have something very similar in yarpc-go (https://github.com/yarpc/yarpc-go/blob/dev/api/transport/handler.go#L31). I was wondering if we could do something similar

Reply:
We could add `MethodType() type` which possibly enumerate to `unary`, `clientStream`, `serverStream` or `bidiStream`